### PR TITLE
[SPARK-27182][SQL] Move the conflict source code of the sql/core module to sql/core/v1.2.1

### PR DESCRIPTION
**Pull request by wangyum**
## What changes were proposed in this pull request?
To make https://github.com/apache/spark/pull/23788 easy to review. This PR moves `OrcColumnVector.java`, `OrcShimUtils.scala`, `OrcFilters.scala` and `OrcFilterSuite.scala` to `sql/core/v1.2.1` and copies it to `sql/core/v2.3.4`.

## How was this patch tested?

manual tests
```shell
diff -urNa sql/core/v1.2.1 sql/core/v2.3.4
```

### CODE REVIEW

**Comment by srowen** [comment]
What are the differences between the copies of these files? is the idea that there is no difference here but your other PR will change the 2.3.4 copy?

**Comment by wangyum** [comment]
Yes. I will change it in other PR. Make other PR easy to review.

**Comment by gatorsmile** [comment]
cc @liancheng 

I am wondering if we can reduce the code duplicate?

**Comment by wangyum** [comment]
Yes. This time revert  `OrcDeserializer` and `OrcSerializer` and moved `checkNoFilterPredicate` to the super class(`OrcTest`).

**Comment by wangyum** [comment]
retest this please

**Comment by wangyum** [comment]
retest this please

**Review by srowen — COMMENTED** [review]
If this is just a code move/copy to set up another PR, and there's no behavior change, and there is no more duplication reduce, looks good to me.

**Comment by HyukjinKwon** [comment]
So do we need https://github.com/apache/spark/pull/24166 before this one?

**Comment by wangyum** [comment]
Yes. Thanks @HyukjinKwon

**Comment by srowen** [comment]
@wangyum https://github.com/apache/spark/pull/24166 is merged now

**Comment by wangyum** [comment]
Thank you @srowen

**Comment by wangyum** [comment]
retest this please

**Comment by maropu** [comment]
retest this please

**Review by srowen — APPROVED** [review]
So this is again basically moving code and making some copies of some code, so that those copies can be modified in a next PR? if so seems OK.

**Comment by liancheng** [comment]
Hey @wangyum, sorry for the long delay. IIUC, this PR is basically a subset of #23788. Once this one is merged, we can simplify #23788 by removing changes made in this PR, right? In that case, this one LGTM.

**Comment by wangyum** [comment]
@liancheng Yes. It's a subset of #23788.

**Review by HyukjinKwon — APPROVED** [review]
Looks fine to me too

File: sql/core/v2.3.4/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilters.scala [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+ *
+ * The workaround employed here is that, for `And`/`Or`/`Not`, we first try to convert their
+ * children with brand new builders, and only do the actual conversion with the right builder
+ * instance when the children are proven to be convertible.
+ *
+ * P.S.: Hive seems to use `SearchArgument` together with `ExprNodeGenericFuncDesc` only.  Usage of
+ * builder methods mentioned above can only be found in test code, where all tested filters are
+ * known to be convertible.
+ */
+private[sql] object OrcFilters extends OrcFiltersBase {
```
Comment (gatorsmile) [inline sql/core/v2.3.4/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilters.scala]: All the functions in this object are different between v2.3.4 and v1.2?

**Comment by gatorsmile** [comment]
I am merging this to master. @wangyum Please submit your follow-up PR for making the actual changes. Thanks!

**Pull request closed without merging**